paddle/fluid/framework/ir/pass.h

```diff
@@ -100,6 +100,10 @@ class Pass {
   // Set a pointer to the attribute. Pass takes ownership of the attribute.
   template <typename AttrType>
   void Set(const std::string &attr_name, AttrType *attr) {
+    if (default_pass_attrs_.count(attr_name) > 0 &&
+        attrs_.count(attr_name) > 0) {
+      attrs_.erase(attr_name);
```

Comment:
I would suggest simply overwrite the attribute value (without erasing it first) and changing the `PADDLE_ENFORCE` below into a log message about updating the attribute value (and putting the log first).

Reply:
Done